src/infer_single_paired.py

```diff
@@ -0,0 +1,262 @@
+    Second member of the tuple corresponds to the result for the
+    second file with possible values- 0,1,2,3,None.
+    (None) if no file provided.
+    Third member of the tuple indicates 4 if the files have split paired
+    end library and 5 if the read id's don't match among the two files.
+
+The modules parses sucessfully for fastq files that are strictly
+as per as wiki convention.
+
+Examples:
+    @HWUSI-EAS100R:6:73:941:1973#0/1
+    @EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG
+
+"""
+import sys
+import gzip
+import re
+from functools import partial
+from enum import Enum
+from Bio.SeqIO.QualityIO import FastqGeneralIterator
+sys.tracebacklimit = 0
+
+
+class Messages(Enum):
+    invalid_file = 0
+    first_mate = 1
+    second_mate = 2
+    mixed_paired_end = 3
+    split_paired_end = 4
+    different_read_ids = 5
+
+
+class EndParser:
+    def __init__(self, file1_name: str, file2_name: str):
+        """Constructor to assign filenames and initialize file pointers.
+
+        Args:
+            file1_name (str) : First file name provided by the user.
+            file2_name (str) : Second file name provide by the user.
+
+        Attributes:
+            file1 : File pointer to first file.
+            file2 : File pointer to second file.
+        """
+        self.file1_name = file1_name
+        self.file2_name = file2_name
+        self.file1 = None
+        self.file2 = None
+
+    def validate(self, seq_id: str):
+        """Validates the type of identifier for the seq_id.
+
+        Args:
+            seq_id (str) : Sequence identifier.
+
+        Returns:
+            int : 0 , 1 , 2
+            Depending on the sequence id whether it matches the
+            wiki conventions it returns 0 for invalid identifier
+            and 1 and 2 for first mate read and second mate read
+            respectively.
+
+        """
+        first_convention = re.compile(r'^[^:#/ ]+:\d+:\d+:\d+:\d+#\d+/[1-2]\s*')
+        second_convention = re.compile(r'^[^:#/ ]+:\d+:[^:#/ ]+:\d+:\d+:\d+:\d+ [1-2]:(Y|N):\d*[02468]:([ATCG]+|\d+)\s*')
+        match_1 = first_convention.finditer(seq_id)
+        match_2 = second_convention.finditer(seq_id)
+        len_match_1 = None
+        len_match_2 = None
+        len_seq_id = len(seq_id)
+        for match in match_2:
+            len_match_2 = match.end() - match.start()
+            break
+        for match in match_1:
+            len_match_1 = match.end() - match.start()
+            break
+        if len_match_1 is not None:
+            if len_seq_id == len_match_1:
+                return 1
+            else:
+                return 0
+        else:
+            if len_match_2 is None:
+                return 0
+            else:
+                if len_seq_id == len_match_2:
+                    return 2
+                else:
+                    return 0
+
+    def check_file1(self):
+        """Opening and checking errors for first file by the user.
+
+        Attribute Error:
```

Comment:
According to the Google-style docstring convention, this should go in a section "Raises:".